// @sparrowengg/twigs-react/dist/src/datepicker/date-field.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useRef } from 'react';
import { useDateFieldState } from 'react-stately';
import { useDateField, useLocale } from 'react-aria';
import { createCalendar } from '@internationalized/date';
import { Box } from '../box';
import { Flex } from '../flex';
import { DateTimeSegment } from './date-time-segment';
export const DateField = (props) => {
    const { locale } = useLocale();
    const state = useDateFieldState({
        ...props,
        locale,
        createCalendar
    });
    const ref = useRef(null);
    const { fieldProps } = useDateField(props, state, ref);
    return (_jsx(Flex, { ...fieldProps, ref: ref, children: state.segments.map((segment, i) => (
        // eslint-disable-next-line  react/no-array-index-key
        _jsx(DateTimeSegment, { segment: segment, state: state }, `twigs-datefield-${i}`))) }));
};
export const StyledField = ({ children, ...otherProps }) => {
    return (_jsx(Box, { ...otherProps, css: {
            transition: 'all 200ms',
            display: 'flex',
            alignItems: 'center',
            p: 0
        }, children: children }));
};
