// @sparrowengg/twigs-react/dist/src/text/text.js
import { jsx as _jsx } from "react/jsx-runtime";
import { styled } from '../../stitches.config';
const StyledText = styled('p', {
    fontWeight: '$4',
    color: '$text',
    variants: {
        size: {
            xxs: {
                fontSize: '$xxs',
                ineHeight: '$xxs'
            },
            xs: {
                fontSize: '$xs',
                ineHeight: '$xs'
            },
            sm: {
                fontSize: '$sm',
                ineHeight: '$sm'
            },
            md: {
                fontSize: '$md',
                lineHeight: '$md'
            },
            lg: {
                fontSize: '$lg',
                lineHeight: '$lg'
            }
        },
        weight: {
            regular: {
                fontWeight: '$4'
            },
            medium: {
                fontWeight: '$5'
            },
            bold: {
                fontWeight: '$7'
            }
        }
    },
    defaultVariants: {
        size: 'sm',
        weight: 'regular'
    }
});
export const Text = ({ children, ...rest }) => {
    return (_jsx(StyledText, { "data-testid": "text", ...rest, children: children }));
};
