// @sparrowengg/twigs-react/dist/src/heading/heading.js
import { jsx as _jsx } from "react/jsx-runtime";
import { styled } from '../../stitches.config';
const StyledHeading = styled('h1', {
    fontWeight: '$7',
    color: '$text',
    variants: {
        size: {
            h6: {
                fontSize: '$lg',
                lineHeight: '$lg'
            },
            h5: {
                fontSize: '$xl',
                lineHeight: '$xl'
            },
            h4: {
                fontSize: '$2xl',
                lineHeight: '$2xl'
            },
            h3: {
                fontSize: '$3xl',
                lineHeight: '$2xl'
            },
            h2: {
                fontSize: '$4xl',
                lineHeight: '$3xl'
            },
            h1: {
                fontSize: '$5xl',
                lineHeight: '$4xl'
            }
        },
        weight: {
            regular: {
                fontWeight: '$4'
            },
            medium: {
                fontWeight: '$5'
            },
            semibold: {
                fontWeight: '$6'
            },
            bold: {
                fontWeight: '$7'
            }
        }
    },
    defaultVariants: {
        size: 'h1',
        weight: 'bold'
    }
});
export const Heading = ({ children, ...rest }) => {
    return (_jsx(StyledHeading, { "data-testid": "heading", ...rest, children: children }));
};
