// @sparrowengg/twigs-react/dist/src/timepicker/timepicker.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useRef } from 'react';
import { useLocale, useTimeField } from 'react-aria';
import { useTimeFieldState } from 'react-stately';
import { Box } from '../box';
import { DateTimeSegment } from '../datepicker/date-time-segment';
import { Flex } from '../flex';
import { FormLabel } from '../form-label';
export const TimePicker = (props) => {
    const { locale } = useLocale();
    const state = useTimeFieldState({
        ...props,
        locale
    });
    const ref = useRef(null);
    const { labelProps, fieldProps } = useTimeField(props, state, ref);
    return (_jsxs(Box, { css: {
            position: 'relative',
            display: 'inline-flex',
            flexDirection: 'column'
        }, children: [props.label
                && _jsx(FormLabel, { ...labelProps, css: { mb: '3px' }, children: props.label }), _jsx(Flex, { ...fieldProps, className: "field", css: {
                    display: 'inline-flex',
                    width: 'auto',
                    background: '$black50',
                    border: 'none',
                    padding: '$4 $6',
                    borderRadius: '$lg'
                }, children: state.segments.map((segment, index) => (_jsx(DateTimeSegment
                // eslint-disable-next-line  react/no-array-index-key
                , { segment: segment, state: state }, `twigs-datetime-segment-${index}`))) })] }));
};
